sdk/src/main/java/dev/cdevents/CDEvents.java

```diff
@@ -116,35 +109,114 @@ public static CDEvent cdEventFromJson(String cdEventJson) {
     }
 
     /**
-     * Validates the cdEventJson against the Schema URL.
-     * @param cdEventJson
-     * @return true, If cdEventJson is valid
+     * Creates a CloudEvent from the custom cdEvent.
+     * @param <T> customCDEvent class
+     * @param customCDEvent custom CDEvent class object of type <T>
+     * @param validateContextSchema true If validation needed against context.schemaUri
+     * @return CloudEvent
      */
-    public static boolean validateCDEventJson(String cdEventJson) {
-        String eventType = getUnVersionedEventTypeFromJson(cdEventJson);
-        CDEventConstants.CDEventTypes cdEventType = getCDEventTypeEnum(eventType);
+    public static <T extends CDEvent> CloudEvent customCDEventAsCloudEvent(T customCDEvent, boolean validateContextSchema) {
+        if (!validateCustomCDEvent(customCDEvent, validateContextSchema)) {
+            throw new CDEventsException("Custom CDEvent validation failed.");
+        }
         try {
-            CDEvent cdEvent = (CDEvent) new ObjectMapper().readValue(cdEventJson, cdEventType.getEventClass());
-            Set<ValidationMessage> errors = getJsonSchemaValidationMessages(cdEvent);
+            return buildCloudEvent(customCDEvent);
+        } catch (URISyntaxException e) {
+            throw new CDEventsException("Exception occurred while building CloudEvent from custom CDEvent ", e);
+        }
+    }
 
-            if (!errors.isEmpty()) {
-                log.error("CDEvent Json validation failed against schema URL {}", cdEvent.schemaURL());
-                log.error("CDEvent Json validation failed with errors {}", errors);
-                return false;
+    /**
+     * Creates customCDEvent from Json string and validates against context and official schemas.
+     * @param customCDEventJson Json string of customCDEvent class type <T>
+     * @param <T> customCDEvent class
+     * @param eventClass custom CDEvent class of type <T>
+     * @param validateContextSchema true If validation needed against context.schemaUri
+     * @return CDEvent
+     */
+    public static <T extends CDEvent> T customCDEventFromJson(String customCDEventJson, Class<T> eventClass, boolean validateContextSchema) {
+        try {
+            T cdEvent = new ObjectMapper().readValue(customCDEventJson, eventClass);
+            if (!validateCustomCDEvent(cdEvent, validateContextSchema)) {
```

Comment:
Same comment above with the reasons something has failed

Reply:
Using an existing method now, and added more logging for the failure reason